captum/optim/_core/loss.py

```diff
@@ -8,37 +10,197 @@
+        return self.__add__(other)
+
+    def __rsub__(self, other: Union[int, float, "Loss"]):
+        return self.__neg__().__add__(other)
+
+    def __rmul__(self, other: Union[int, float, "Loss"]):
+        return self.__mul__(other)
+
+    def __rtruediv__(self, other: Union[int, float, "Loss"]):
+        if isinstance(other, (int, float)):
+
+            def loss_fn(module):
+                return operator.truediv(other, torch.mean(self(module)))
+
+            name = self.__name__
+            target = self.target
+        elif isinstance(other, Loss):
+            # This should never get called because __div__ will be called instead
+            pass
+        else:
+            raise TypeError(
+                "Can only apply math operations with int, float or Loss. Received type "
+                + str(type(other))
+            )
+        return CompositeLoss(loss_fn, name=name, target=target)
+
+    def __rpow__(self, other: Union[int, float, "Loss"]):
+        if isinstance(other, (int, float)):
+
+            def loss_fn(module):
+                return operator.pow(other, torch.mean(self(module)))
+
+            name = self.__name__
+            target = self.target
+        elif isinstance(other, Loss):
+            # This should never get called because __pow__ will be called instead
+            pass
+        else:
+            raise TypeError(
+                "Can only apply math operations with int, float or Loss. Received type "
+                + str(type(other))
+            )
+        return CompositeLoss(loss_fn, name=name, target=target)
+
+
+def module_op(self: Loss, other: Union[None, int, float, Loss], math_op: Callable):
+    """
+    This is a general function for applying math operations to Losses
+    """
+    if other is None and math_op == operator.neg:
+
+        def loss_fn(module):
+            return math_op(self(module))
+
+        name = self.__name__
+        target = self.target
+    elif isinstance(other, (int, float)):
+
+        def loss_fn(module):
+            return math_op(self(module), other)
+
+        name = self.__name__
+        target = self.target
+    elif isinstance(other, Loss):
+        # We take the mean of the output tensor to resolve shape mismatches
+        def loss_fn(module):
+            return math_op(torch.mean(self(module)), torch.mean(other(module)))
+
+        name = f"Compose({', '.join([self.__name__, other.__name__])})"
+        target = (
+            self.target if hasattr(self.target, "__iter__") else [self.target]
+        ) + (other.target if hasattr(other.target, "__iter__") else [other.target])
+    else:
+        raise TypeError(
+            "Can only apply math operations with int, float or Loss. Received type "
+            + str(type(other))
+        )
+    return CompositeLoss(loss_fn, name=name, target=target)
+
+
+class BaseLoss(Loss):
+    def __init__(self, target: nn.Module = [], batch_index: Optional[int] = None):

```

Comment:
You should add the missing the `-> None` type hint as specified by PEP-0484 here.